packages/ragbits-cli/src/ragbits/cli/__init__.py

```diff
@@ -1,12 +1,30 @@
 import importlib.util
 import pkgutil
 from pathlib import Path
+from typing import Annotated
 
-from typer import Typer
+import typer
 
 import ragbits
 
-app = Typer(no_args_is_help=True)
+from .app import CLI, OutputType
+
+app = CLI(no_args_is_help=True)
+
+
+@app.callback()
+def output_type(
+    output: Annotated[
+        OutputType, typer.Option("--output", "-o", help="Set the output type (text or json)")
+    ] = OutputType.text.value,  # type: ignore
+) -> None:
+    # type ignore is a workaround for typer issue
+    # enforcing to use str as default value
```

Comment:
```suggestion
    # `OutputType.text.value` used as a workaround for the issue with `typer.Option` not accepting Enum values
    output: Annotated[
        OutputType, typer.Option("--output", "-o", help="Set the output type (text or json)")
    ] = OutputType.text.value,  # type: ignore
) -> None:
```

The workaround is providing `OutputType.text.value` instead of the properly typed enum value. Type ignore is just shutting mypy up about it.